tests/integration/test_docker_builds.py

```diff
@@ -392,6 +392,86 @@ def test_exact(tmpdir):
             cli.remove_image(image, force=True)
 
 
+def test_dirty_fails_without_flag(tmpdir):
+    tmp = tmpdir.join('shipwright-sample')
+    path = str(tmp)
+    source = pkg_resources.resource_filename(
+        __name__,
+        'examples/shipwright-sample',
+    )
+    create_repo(path, source)
+    tmp.join('service1/base.txt').write('Some text')
+    client_cfg = docker_utils.kwargs_from_env()
+
+    cli = docker.Client(version='1.18', **client_cfg)
+
+    args = get_defaults()
+
+    try:
+        with pytest.raises(SystemExit):
+            shipw_cli.run(
+                path=path,
+                client_cfg=client_cfg,
+                arguments=args,
+                environ={},
+            )
+
+    finally:
+        old_images = (
+            cli.images(name='shipwright/base', quiet=True)
+        )
+        for image in old_images:
+            cli.remove_image(image, force=True)
+
+
+def test_dirty_flag(tmpdir):
+    tmp = tmpdir.join('shipwright-sample')
+    path = str(tmp)
+    source = pkg_resources.resource_filename(
+        __name__,
+        'examples/shipwright-sample',
+    )
+    create_repo(path, source)
+    tmp.join('service1/base.txt').write('Some text')
+    client_cfg = docker_utils.kwargs_from_env()
+
+    cli = docker.Client(version='1.18', **client_cfg)
+
+    args = get_defaults()
+    args['--dirty'] = True
+
+    try:
+        shipw_cli.run(
+            path=path,
+            client_cfg=client_cfg,
+            arguments=args,
+            environ={},
+        )
+
+        service1, shared, base = (
+            cli.images(name='shipwright/service1') +
+            cli.images(name='shipwright/shared') +
+            cli.images(name='shipwright/base')
+        )
+
+        assert 'shipwright/service1:latest' in service1['RepoTags']
```

Comment:
It should be possible to get something like:
```python

        service1diffhash = ...;

         assert set(service1['RepoTags']) == {
             'shipwright/service1:master',
             'shipwright/service1:master',
             'shipwright/service1:latest',
             'shipwright/service1:latest',
             'shipwright/service1:' + tag + '-dirty-' + service1diffhash,
         }
```

also I suspect that the diffhashes are always the same, because they don't take into account timestamps? They're only blob hashes.